airflow/providers/amazon/aws/sensors/bedrock.py

```diff
@@ -68,7 +70,56 @@ def __init__(
         super().__init__(**kwargs)
         self.deferrable = deferrable
 
-    def poke(self, context: Context) -> bool:
+    def poke(self, context: Context, **kwargs) -> bool:
+        state = self.get_state()
+        if state in self.FAILURE_STATES:
+            # TODO: remove this if block when min_airflow_version is set to higher than 2.7.1
+            if self.soft_fail:
+                raise AirflowSkipException(self.FAILURE_MESSAGE)
+            raise AirflowException(self.FAILURE_MESSAGE)
+
+        return state not in self.INTERMEDIATE_STATES
+
+    @abc.abstractmethod
+    def get_state(self) -> str:
+        """Implement in subclasses."""
+
+
```

Comment:
@Taragolis  - These two base sensors are identical other than the aws_hook_class (BedrockHook vs BedrockAgentHook) and the hook name in the parent class.  I'm not really familiar with how you did the `AwsBaseSensor[BedrockAgentHook]` thing, is it possible to combine these into one BaseBedrockSensor?  What would that look like?

Reply:
If you would like to make generic from the generic you could do something like that

```python

_GenericBedrockHook = TypeVar("_GenericBedrockHook", BedrockAgentHook, BedrockHook)

class BedrockBaseSensor(AwsBaseSensor[_GenericBedrockHook]):
    """
    General sensor behavior for Amazon Bedrock.

    Subclasses must implement following methods:
        - ``get_state()``

    Subclasses must set the following fields:
        - ``INTERMEDIATE_STATES``
        - ``FAILURE_STATES``
        - ``SUCCESS_STATES``
        - ``FAILURE_MESSAGE``

    :param deferrable: If True, the sensor will operate in deferrable mode. This mode requires aiobotocore
        module to be installed.
        (default: False, but can be overridden in config file by setting default_deferrable to True)
    """

    INTERMEDIATE_STATES: tuple[str, ...]
    FAILURE_STATES: tuple[str, ...]
    SUCCESS_STATES: tuple[str, ...]
    FAILURE_MESSAGE: str

    ui_color = "#66c3ff"
    aws_hook_class: type[_GenericBedrockHook]

    ...

class BedrockCustomizeModelCompletedSensor(BedrockBaseSensor[BedrockHook]):
    aws_hook_class = BedrockHook
    ...

class BedrockKnowledgeBaseActiveSensor(BedrockBaseSensor[BedrockAgentHook]):
    aws_hook_class = BedrockAgentHook
    ...
```

Reply:
AH!  I see, thanks.    I think I like the idea, but maybe I'll do that in a separate PR.  What do you think?  Does it feel worth doing?  Does it have to be now or can we let it stand how it is for now?

Reply:
I do not have personal preferences, so it could be done by the separate PR or even keep as it is.